### PR TITLE
Issue #4384: fix JavadocParagraph violation in JavadocTokenTypes

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue #4384

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java [hunk 1]
```diff
@@ -39,7 +39,7 @@
      * Root node of any Javadoc comment.
      * Last child is always {@link #EOF}.
      *
-     * <p>
+     * <p/>
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java]: Why do we have an empty `p` tag? If this is to enforce new lines in html/report, shouldn't we use `<br />` or surround whole text in `<p>`.

In this case, I think we should surround `<b>Tree for example:</b>` with `<p>`.
=> `<p><b>Tree for example:</b></p>`

Comment (Nimfadora) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java]: [Here](http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#format) it is said that we should separate paragraphs with `<p>` tag. But if we don't care on this old document, I would rather rewrite it as you have said: `<p><b>Tree for example:</b></p>`, as `<br/>` tag is also old-fashioned solution.

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java]: @romani Can confirm, but in xdocs we write paragraphs surrounding the entire text. I feel we should follow suite in the Javadocs.
Ex: https://github.com/checkstyle/checkstyle/blob/fd7fb45d409a2de0dead15f907e69b330dbaa5dc/src/xdocs/config_blocks.xml#L26-L38

> Here it is said that we should separate paragraphs with <p> tag.

To me this also looks like the old way of writing HTML before XHTML came and standardized the layout. A new PR will also prevent some Javadoc checks from executing on unstructured Javadocs so this is another reason why should avoid this way of writing.

Comment (Nimfadora) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java]: fixed

**Comment by Nimfadora** [comment]
@rnveach TC failed on timeout:
```
[19:50:48]Step 2/2: Inspections (IntelliJ IDEA) (13m:07s)
[20:03:53][Step 2/2] The build Checkstyle::IDEA Inspections Pull Request #7283 {buildId=1154747} has been running for more than 15 minutes. Terminating...
[20:03:53][Step 2/2] Error message is logged
[20:03:53][Step 2/2] Execution timeout
[20:03:56][Step 2/2] Process exited with code 137
```
What should we do?

**Comment by rnveach** [comment]
@Nimfadora I restarted TC.
That is all you need to request of us.

@romani We keep having to restart TC due to timeout. Should we report issue?

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java]: confirmed.
format: `<p>...</p>` is expected. in future this Check will work on XHTML-like format, and all of our javadocs will become XHTML-like. Event it is not favorite form for HTML.

**Comment by romani** [comment]
> We keep having to restart TC due to timeout. Should we report issue?

yes, looks like they upgraded IDEA version and it does not work anymore.

**Pull request merged**